src/js/utils/parser.js

```diff
@@ -54,6 +54,20 @@ define([
         return '';
     });
 
+    parser.getMediaId = function(url) {
+        var filenameRegex = /[^\/]+$/;
+        var mediaIdRegex = /[^.-]*/;
```

Comment:
Let's try to use one RegEx. For example this would match the video urls we know of (See examples in next comment):
```js
/.*\/(?:manifests|videos)\/([a-zA-Z0-9]{8})[\.-].*/
```